src/experiment_handler.py

```diff
@@ -216,61 +211,54 @@ def update_experiment():
         #                               "Need completed data")
 
         #data = socket_operations.recv_data(exp_handler_to_dsp,
-        #                             options.dsp_to_exphan_identity, printing)
+        #                             options.dsp_to_exphan_identity, log)
 
         some_data = None  # TODO get the data from data socket and pass to update
 
         semaphore.acquire()
         change_flag = exp.update(some_data)
         if change_flag:
-            if __debug__:
-                printing("Building an updated experiment.")
+            log.debug("building an updated experiment")
             exp.build_scans()
-            printing(f"Experiment {exp.__class__.__name__} with CPID {exp.cpid} successfully updated")
+            log.info("experiment successfully updated",
+                     experiment_name=exp.__class__.__name__,
+                     cpid=exp.cpid)
         semaphore.release()
 
     update_thread = threading.Thread(target=update_experiment)
 
     while True:
-
         if not change_flag:
             serialized_exp = pickle.dumps(None, protocol=pickle.HIGHEST_PROTOCOL)
         else:
             exp.build_scans()
-            printing(f"Successful experiment {exp.__class__.__name__} built with CPID {exp.cpid}")
+            log.info("experiment successfully built",
+                     experiment_name=exp.__class__.__name__,
+                     cpid=exp.cpid)
             serialized_exp = pickle.dumps(exp, protocol=pickle.HIGHEST_PROTOCOL)
-            # use the newest, fastest protocol (currently version 4 in python 3.4+)
+            # Use the newest, fastest protocol (currently version 4 in python 3.4+)
             change_flag = False
 
-        # WAIT until radar_control is ready to receive a changed experiment
+        # Wait until radar_control is ready to receive a changed experiment
         message = socket_operations.recv_request(exp_handler_to_radar_control,
                                                  options.radctrl_to_exphan_identity,
-                                                 printing)
-        if __debug__:
-            request_msg = f"Radar control made request -> {message}."
```

Comment:
How come this message was deleted?

Reply:
I added a log.debug() for it. I didn't feel it was useful to log the message in DEBUG and then directly after always log the same message if EXPNEEDED was received.